src/main/java/net/atos/entng/support/helpers/UserInfosHelper.java

```diff
@@ -83,11 +95,23 @@ public static UserInfos.Action getUserActionFromJSON(JsonObject oAction) {
         return action;
     }
 
+    public static UserInfos.Application getUserAppsFromJSON(JsonObject oApplication) {
+        UserInfos.Application application = new UserInfos.Application();
+        application.setAddress(oApplication.getString(Ticket.ADDRESS));
+        application.setDisplayName(oApplication.getString(Ticket.DISPLAYNAME));
+        return application;
+    }
+
      @SuppressWarnings("unchecked")
      public static List<UserInfos.Action> getUserActionsFromJSONArray(JsonArray actions) {
         return ((List<JsonObject>) actions.getList()).stream().map(UserInfosHelper::getUserActionFromJSON).collect(Collectors.toList());
      }
 
+    @SuppressWarnings("unchecked")
+     public static List<UserInfos.Application> getUserAppsFromJSONArray(JsonArray applications) {
+        return ((List<JsonObject>) applications.getList()).stream().map(UserInfosHelper::getUserAppsFromJSON).collect(Collectors.toList());
+     }
```

Comment:
et tu vas me dire que tu n'avais pas vu que tu pouvais mutualiser ton code ?? 😏😏
```suggestion
@SuppressWarnings("unchecked")
public static <T> List<T> getObjectsFromJsonArray(JsonArray jsonArray, Function<JsonObject, T> mapper) {
    return ((List<JsonObject>) jsonArray.getList()).stream().map(mapper).collect(Collectors.toList());
}
```
après tu peux faire:
List<UserInfos.Application> userApps = getObjectsFromJSONArray(applications, UserInfosHelper::getUserAppsFromJSON);